src/frame_clock.rs

```diff
@@ -5,13 +5,18 @@
 use gdk_sys;
 use glib::translate::*;
 use FrameClock;
+use glib::object::IsA;
 
-impl FrameClock {
-    pub fn get_refresh_info(&self, base_time: i64) -> (i64, i64) {
+pub trait FrameClockExtManual: 'static {
```

Comment:
Same

Reply:
Then we need remove trait from generated code too

Reply:
Interesting, why is it generated at all? It shouldn't because the configuration is not overridden, and the default behaviour is to only generate the trait if there is a class struct (-> it is possible to create subclasses outside this crate) or there are known subclasses (-> there are subclasses inside this crate).

Reply:
There class for FrameClock, but this class is disguised and `mark_final_types` don't checks that
```<record name="FrameClockClass" c:type="GdkFrameClockClass" disguised="1" glib:is-gtype-struct-for="FrameClock">```

Reply:
Ah we need to fix that in gir then

Reply:
I tried fix that, but found this comment in parser.
Still think that to do.
```
        // Currently g-ir-scanner is too eager to mark all typedef to non-complete
        // types as disguised, so we limit this behaviour to the two most common cases.
        // https://gitlab.gnome.org/GNOME/gobject-introspection/issues/101
        let disguised = c_type == "GdkAtom" || c_type == "GIConv";
```

Reply:
I meant only ignoring the disguised ones for the class detection code

Reply:
Problem that this is place in parser where disguised  must be parsed.
But problem seems solved and we can use disguised="1".
I will add issue to collect all PRs

Reply:
https://github.com/gtk-rs/gir/issues/760